lib/enum/value.rb

```diff
@@ -0,0 +1,143 @@
+module Enum
+  class Value
+    
+    # TODO: Perhaps stored_value should be renamed unsafe_value?
+    
+    include Comparable
+
+    attr_reader :stored_value, :error
+    
+    class << self
+      attr_accessor :default_value, :suppress_read_errors, :klass
+    end
+    
+    def self.inherited(subclass)
+      # Value subclass settings and options
+      #   @default_value => <:ERROR|:ANY|:something|nil>
+      #   @suppress_read_errors => <true|false>
+      subclass.suppress_read_errors = false
+      subclass.default_value = :ERROR
+    end
+    
+    # Combined getter/setter for 'default_value'.
+    def self.default_value(*args)
+      # Can't use .any? because [nil].any? is false (ruby 2.4),
+      # and nil is a valid argument here.
+      # [nil].empty? is also false, which is what we want.
+      if !args.empty?
+        @default_value = args[0]
+      else
+        @default_value
+      end
+    end
+    
+    # Combined getter/setter for 'suppress_read_errors'.
+    def self.suppress_read_errors(*args)
+      if !args.empty?
+        @suppress_read_errors = args[0]
+      else
+        @suppress_read_errors
+      end
+    end
+    
+    # Load a primitive (symbol, string, integer) into new enum Value instance,
+    # taking into consideration enum constraints, default_value setting.
+    # Returns frozen Value instance.
+    def initialize(new_val) #, enum_class)
```

Comment:
remove commented code

Reply:
~~~new_val~~~`raw_val`? - it's not a new value, it's those one that is passed outside, and it's done once, right?

Reply:
I agree, `raw_val` is a better name for that variable. I'll change that and clean up the comments.